gst/nnstreamer/elements/gsttensor_trainer.c

```diff
@@ -702,25 +819,16 @@ gst_tensor_trainer_sink_event (GstPad * sinkpad, GstObject * parent,
     GstEvent * event)
 {
   GstTensorTrainer *trainer;
-  GstTensorTrainerFrameworkInfo info;
   trainer = GST_TENSOR_TRAINER (parent);
 
   GST_DEBUG_OBJECT (trainer, "Received %s event: %" GST_PTR_FORMAT,
       GST_EVENT_TYPE_NAME (event), event);
 
   switch (GST_EVENT_TYPE (event)) {
     case GST_EVENT_EOS:
-      trainer->fw->getFrameworkInfo (trainer->fw, NULL, trainer->privateData,
-          &info);
-      if (!info.is_training_complete) {
-        GST_INFO_OBJECT (trainer,
-            "got GST_EVENT_EOS event but training is not completed, state is %d",
-            GST_STATE (trainer));
-        g_mutex_lock (&trainer->trainer_lock);
-        GST_INFO_OBJECT (trainer, "wait for training_complete_cond signal");
-        g_cond_wait (&trainer->training_complete_cond, &trainer->trainer_lock);
-        g_mutex_unlock (&trainer->trainer_lock);
-      }
+      if (gst_tensor_trainer_wait_for_training_completion (trainer))
+        GST_DEBUG_OBJECT (trainer, "training is completed in sub-plugin[%s]",
```

Comment:
If you need ```fw_name``` frequently, just make a inline function.

Reply:
Yes, I will consider it for the next PR if needed.